### ISSUE TITLE
Flang doesn't warn about use of unsupported CHARACTER KIND=2

### ISSUE THREAD

**Issue by nSircombe** [flang-compiler/flang#864]
Flang's default character set is ASCII, which corresponds to KIND=1.

This appears to be the only supported CHARACTER KIND in Flang (https://github.com/flang-compiler/flang/blob/cb42a171b6462b2f3acd3e1841a56a1ae08291ef/runtime/flang/miscsup_com.c).

Attempting to use other KINDs should generate an error at compile time. For example:

```
PROGRAM charkind
 IMPLICIT NONE                                                                                                                                                                                                                                
 CHARACTER(KIND=2, LEN=4):: data
END PROGRAM charkind
```

Here KIND=2 corresponds, I think, to JIS X 0213. This generates no error at compile time, where as KIND=4 (ISO 10646) does generate an error:
```
F90-S-0081-Illegal selector - KIND parameter has unknown value for data type  (charkind.f90: 5)
  0 inform,   0 warnings,   1 severes, 0 fatal for charkind
```

This can lead to build systems returning a false positive for CHARACTER(KIND=2) support.
